src/components/CTA.tsx

```diff
@@ -1,22 +1,36 @@
 import { useEffect, useState } from "react";
 
-export function CTA() {
+type ResponseData = {
+  success: boolean;
+};
+
+type CTAProps = {
+  redirect: (path: string) => Promise<void>;
+};
+
+export function CTA({ redirect }: CTAProps) {
   const handleSubmit = async (event) => {
     event.preventDefault();
 
-    // Perform some action before submitting the form
-    // Update count in database
-    const resp = await fetch("/updateWaitlistTotal.json", {
-      method: "POST",
-    });
+    try {
+      const res = await fetch("/signup.json", {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({ email }),
+      });
+      const data = (await res.json()) as ResponseData;
 
-    // Submit the form
-    event.target.submit();
-  };
-  type ResponseData = {
-    totalCount: number;
+      if (data.success) {
+        // help
+        await redirect("/thank-you");
```

Comment:
@coding-in-public do you know how to redirect to another route in Astro in a React component? I have spent 45mins trying lol and cannot figure it out. 

what i've tried:
- passing Astro.redirect as a prop to my React component
- calling redirect in my post API endpoint
- window.location.href
- windows.location.push

Reply:
Hmm…@jsjoeio you should be able to just replace the URL, no? like this…

 location.replace("/thank-you");

Reply:
Ah thank you! Got it working :)

Reply:
This was the issue FWIW: https://twitter.com/jsjoeio/status/1651634164788248576?s=20